modules/commandModule/commandWorker_flight.py

```diff
@@ -1,18 +1,56 @@
-from modules.commandModule.commandFns import write_pigo
+from modules.commandModule.commandModule import CommandModule
+from modules.commandModule.commandFns import write_pigo, read_pogi
 
-def flight_command_worker(pipelineIn, pipelineOut):
-	# pipelineIn gives [[x,y], [range]]
-	if pipelineIn.empty:
-	   continue
-	
-	data = pipelineIn.get()
-	# Cache data here
-	newPigo = {
-		'gpsCoordinates': [x, y]
-	}
+POGI_DIR = ""
+PIGO_DIR = ""
 
-	write_pigo(newPigo)
+def pogi_subworker(pipelineOut, POGI_DIR):
 
-	#TODO: Write logic to get POGI
+    # get the pogi data
+    changed, pogiData = read_pogi(POGI_DIR)
 
-	
\ No newline at end of file
+    # if pogi has changed, output to the pipeline
+    if changed:
+        pipelineOut.put(pogiData)
+
+
+def flight_command_worker(pipelineIn, pipelineOut, pause, exitRequest):
+    
+    """
+    Worker function for PIGO and POGI data transactions
+
+    Parameters
+    ----------
+    pipelineIn : multiprocessing.Queue
+        input pipeline for pigoData
+    pipelineOut : multiprocessing.Queue
+        output pipeline for pogiData
+    pause : multiprocessing.Lock
+        a lock shared between worker processes
+    exitRequest : multiprocessing.Queue
+        a queue that determines when the worker process stops
+    """
+
+    while True:
+
+        pause.acquire()
+        pause.release()
+        
+        # PIGO Logic
+        # pipelineIn gives [[x,y], [range]]
+        if pipelineIn.empty:
+            continue
+        
+        data = pipelineIn.get()
```

Comment:
The only slight issue with using one worker for both POGI and PIGO is that if there is no incoming telemetry, we'll be stuck at this line, and won't be able to send any commands to the plane until we get a telemetry reading, which isn't bad I suppose because:
1. We'll most likely always be getting a telemetry reading
2. If we're not, we probably won't be able to uplink anyway

But I need a thumbs up from @Shrinjay on this, because otherwise we'll have to separate POGI and PIGO into their own separate workers.

Reply:
If Shrinjay gives a thumbs up, no need to make a new worker